src/transaction/TransactionController.ts

```diff
@@ -1114,23 +1136,24 @@ export class TransactionController extends BaseController<
       etherscanTokenResponse,
     ] = await handleTransactionFetch(networkType, address, opt);
 
-    const normalizedTxs = etherscanTxResponse.result.map(
+    let normalizedTxs = etherscanTxResponse.result.map(
       (tx: EtherscanTransactionMeta) =>
         this.normalizeTx(tx, currentNetworkID, currentChainId),
     );
-    const normalizedTokenTxs = etherscanTokenResponse.result.map(
+    let normalizedTokenTxs = etherscanTokenResponse.result.map(
       (tx: EtherscanTransactionMeta) =>
         this.normalizeTokenTx(tx, currentNetworkID, currentChainId),
     );
 
-    const remoteTxs = [...normalizedTxs, ...normalizedTokenTxs].filter((tx) => {
-      const alreadyInTransactions = this.state.transactions.find(
-        ({ transactionHash }) => transactionHash === tx.transactionHash,
-      );
-      return !alreadyInTransactions;
-    });
+    normalizedTxs = this.trimTransactionsForState(normalizedTxs);
```

Comment:
Interesting. Why are transactions being trimmed at this point? This is a bit of a confusing usage of this function, since these aren't set to state in this case.

Reply:
The intention of this is to avoid processing a large array from Etherscan, since their endpoint could return a max of 10.000 txs, we could end up, in a worst case scenario, with a 20.000 txs (between eth and token txs) array and the `fetchAll` that handles this array is called with a high frequency. Do you think there's a better way to handle this?

Reference: https://docs.etherscan.io/api-endpoints/accounts

`Note : This API endpoint returns a maximum of 10000 records only.`

Reply:
The `offset` parameter can be used to reduce the number of records returned by that API. If we use `offset`, the data doesn't get sent back to us at all, so we save on network bandwith and processing time as well.

Reply:
Are you sure it's the `offset`? I think the `limit` helps with that and the `offset` sets starting point. I had forgotten about this 🤦

Reply:
The docs seem to indicate that it's `offset`. There is no documented `limit` parameter. I haven't tried it yet though

Reply:
Fixed! Thanks for the help @Gudahtt